torchtitan/components/tokenizer.py

```diff
@@ -25,3 +31,333 @@ def decode(self, *args, **kwargs) -> str:
     @property
     def n_words(self) -> int:
         return self._n_words
+
+
+class TorchTitanTokenizer(Tokenizer):
```

Comment:
```suggestion
class Tokenizer(BaseTokenizer):
```
or `HuggingFaceTokenizer` similar to https://github.com/pytorch/torchtitan/blob/main/torchtitan/datasets/hf_datasets.py#L71